src/osx/core/hidjoystick.cpp

```diff
@@ -861,6 +861,10 @@ void* wxJoystickThread::Entry()
         //is the cookie a button?
         if (nIndex < 40)
         {
+            // Bounds check for bit shift in this block.
+            wxCHECK_MSG(nIndex >= 0, (void *) -1, "Shift count negative.");
```

Comment:
Sorry, I don't understand how does this `(void*)-1` compile: this is a void function, so it can't return anything, please use `wxCHECK_RET()` here (and just below).

Also, `nIndex` here seems very clearly positive, so I'm not sure if it's worth having this one at all...

Reply:
OK. I removed the check.